Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -238,332 +159,134 @@ public final class OAuth2Client {
-                                                      completion: @escaping (Result<APIResponse<Token>, APIClientError>) -> Void)
-    {
-        guard case let .success(response) = oauthTokenResponse else {
-            completion(oauthTokenResponse)
-            return
-        }
-        
-        // Retrieves the org's OpenID configuration
-        self.openIdConfiguration { result in
-            switch result {
-            case .failure(let error):
-                completion(.failure(.serverError(error)))
-            case .success:
-                do {
-                    try response.result.validate(using: self, with: request.tokenValidatorContext)
-                } catch {
-                    completion(.failure(.validation(error: error)))
-                    return
-                }
-                
-                guard let idToken = response.result.idToken else {
-                    completion(oauthTokenResponse)
-                    return
-                }
-                
-                guard let keySet = keySet else {
-                    completion(.failure(.validation(error: JWTError.invalidKey)))
-                    return
-                }
-                
-                do {
-                    if try idToken.validate(using: keySet) == false {
-                        completion(.failure(.validation(error: JWTError.signatureInvalid)))
-                        return
-                    }
-                } catch {
-                    completion(.failure(.validation(error: error)))
-                    return
-                }
-                completion(oauthTokenResponse)
-            }
-        }
+        return response
     }
 
     // MARK: Private properties / methods
     private let delegates = DelegateCollection<OAuth2ClientDelegate>()
-
-    private let refreshLock = Lock()
     private(set) lazy var refreshQueue: DispatchQueue = {
         DispatchQueue(label: "com.okta.refreshQueue.\(baseURL.host ?? "unknown")",
                       qos: .userInitiated,
                       attributes: .concurrent)
     }()
 
-    private let configurationLock = Lock()
-    private lazy var configurationQueue: DispatchQueue = {
-        DispatchQueue(label: "com.okta.configurationQueue.\(baseURL.host ?? "unknown")",
-                      qos: .userInitiated,
-                      attributes: .concurrent)
-    }()
-    internal var openIdConfigurationAction: CoalescedResult<Result<OpenIdConfiguration, OAuth2Error>>?
-
-    private lazy var jwksQueue: DispatchQueue = {
-        DispatchQueue(label: "com.okta.jwksQueue.\(baseURL.host ?? "unknown")",
-                      qos: .userInitiated,
-                      attributes: .concurrent)
-    }()
-    internal var jwksAction: CoalescedResult<Result<JWKS, OAuth2Error>>?
+    let openIdConfigurationAction = CoalescedResult<OpenIdConfiguration>(taskName: "OpenIdConfiguration")
+    let jwksAction = CoalescedResult<JWKS>(taskName: "OpenIdConfiguration")
```

Comment:
Whoops, I need to change this task name

Reply:
I fixed this [in the follow-up PR](https://github.com/okta/okta-mobile-swift/pull/228).